File: dataset/process_emb.py
import numpy as np

def load_emb(src):
	with open(src) as f:
		lines = f.readlines()
	total, size = lines[0].strip().split(' ')
	total = int(total); size = int(size)
	print('total: %d, embedding size: %d' % (total, size))
	del(lines[0])

	words = []; embs = []
	for l in lines:
		items = [ item for item in l.strip().split(' ') if item ]
		assert(len(items) == size + 1)
		words.append(items[0])
		embs.append(np.array([float(item) for item in items[1:]]))
	if '*unk*' not in words:
		mean = np.zeros(size)
		for e in embs:
			mean += e
		mean = mean / len(embs)
		words.append('*unk*')
		embs.append(mean)
	if '*pad*' not in words:
		words.append('*pad*')
		embs.append(np.zeros(size))
	assert(len(words) == len(embs))
	return words, np.array(embs)

def convert(src, dst_dir):
	words, embs = load_emb(src)
	identity = src.rsplit('/', 1)[-1]
	with open('%s/%s_words.txt' % (dst_dir, identity), 'w') as f:
		f.write('\n'.join(words)+'\n')
	np.save('%s/%s_embs.npy' % (dst_dir, identity), embs)

def process():
	todo = ['CBOW_50', 'CBOW_100', 'CBOW_200', 'CBOW_300', 'CBOW_500',
		'SG_50', 'SG_100', 'SG_200', 'SG_300', 'SG_500', 'HLBL_50']
	for f in todo:
		convert('word_embedding/origin/'+f, 'word_embedding/')

if __name__ == '__main__':
	process()

File: dataset/process_emb.lua
local npy4th = require 'npy4th'
local util = require 'util/util'

local function convert(words_src, embs_src, dst)
	local f = io.open(words_src, 'r')
	local words = util.split_line(f:read('*all'))
	f:close()
	
	local word_to_index = {}
	for i, w in ipairs(words) do
		word_to_index[w] = i
	end
	print(embs_src)	
	local embs = npy4th.loadnpy(embs_src)
	torch.save(dst, {
		word_to_index = word_to_index,
		index_to_emb = embs
	})
end

local function process()
	local src_dir = 'word_embedding'
	local fnames = {'CBOW_50', 'CBOW_100', 'CBOW_200', 'CBOW_300',
		'CBOW_500', 'SG_50', 'SG_100', 'SG_200', 'SG_300', 'SG_500',
		'HLBL_50'}

	for _, fn in ipairs(fnames) do
		local words_src = string.format('%s/%s_words.txt', src_dir, fn)
		local embs_src = string.format('%s/%s_embs.npy', src_dir, fn)
		local dst = string.format('%s/%s.t7', src_dir, fn)
		convert(words_src, embs_src, dst)
	end
end

do
	process()
end
